### ISSUE TITLE
More `is_*` functions

### ISSUE THREAD

**Issue by tinkerer-red** [YoYoGames/GameMaker-Feature-Requests#1332]
### Is your feature request related to a problem?

`is_constructor` is what started my journey down this road.
The only way to find out if a function is a constructor would be to do the following:
```gml
function is_constructor(_func){
	try {
		var _temp = new _func();
		delete _temp;
		
		return true;
	}
	catch (err) {
		if (string_pos("constructor", err.message))
		|| (string_pos("'new'", err.message)) {
			return false;
		}
		return true;
	}
}
```

Which obviously isnt practical.

Adding to this `is_object()` is another function which would be useful in the event you need to differentiate structs and objects, like when you are building on the debug tools and you wish to log built in variables (which dont appear in a `structforeach` or `struct_get_names`)
```gml
function is_object(_obj) {
    if (!is_struct(_obj))
    && (typeof(_obj) == "struct") {
        return true;
    }
    return false;
}
```

### Describe the solution you'd like

more `is_*` functions

**Comment by Alphish** [comment]
For the record, I already made a feature request for `is_constructor` and `constructor_call` in feature request #1225.

I wonder about `is_object` thing; I'd actually call it `is_instance`, because "object" is more associated with object assets. More importantly, if you pass an instance ID, its type will be "ref" instead. At any rate, from my testing it seems that `instance_exists` covers both instance IDs and instance self-structs?

**Comment by KormexGameMaker** [comment]
I'd love `is_tilemap`, for use with the collision list functions and sending an array of both objects and tilemaps

**Comment by Gamer-XP** [comment]
From my testing, instance_exists returns true even if you throw some random struct at it. Checking if given thing is an instance or not is kind hacky right now.

**Comment by JordOfTheFlies** [comment]
Some flavour of this for deactivated instances would be nice too, there are a couple of ways to detect this manually (such as checking if variable_instance_names_count(id) == -1), but it'd be nice to have a safer official way of telling if a valid id corresponds to an activate or inactive instance.

**Comment by BackYard321** [comment]
> From my testing, instance_exists returns true even if you throw some random struct at it. Checking if given thing is an instance or not is kind hacky right now.

I agree - the best solution I could come up with is converting the ref to a string, and checking if that string contains the substring "instance" - not great!
```gml
function is_instance(_variable)
{
	return is_handle(_variable)
	&& string_pos("instance", string(_variable)) != 0
}
```

**Comment by Gamer-XP** [comment]
> > From my testing, instance_exists returns true even if you throw some random struct at it. Checking if given thing is an instance or not is kind hacky right now.
> 
> I agree - the best solution I could come up with is converting the ref to a string, and checking if that string contains the substring `"instance"`. Not great!
> 
> ```c++
> function is_instance(_variable)
> {
> 	return is_handle(_variable)
> 	&& string_pos("instance", string(_variable)) != 0
> }
> ```

I think, easier solution will be just to try reading some instance-only field inside of try/catch. I've made this for different purpsoe, but it should also work to check if given thing is an instance:

```
/**
	Returns true when instance exists even if it was deactivated
*/
function instance_exists_complete( _instance )
{
	if instance_exists(_instance)
		return true;
	
	try
	{
		var check = _instance.id;
		return true;
	}
	catch(e)
	{
		return false;
	}
}
```
